### ISSUE TITLE
Issue with tf.nn.softmax_cross_entropy_with_logits when label value is 100+

### ISSUE THREAD

**Issue by Raj-08** [tensorflow/tensorflow#29077]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: YES
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:  Installed from Docker official repo
- **TensorFlow version (use command below)**: Ver 1.12.0
- **Python version**: python 3.6
- **Bazel version (if compiling from source)**: 
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:  7.4
Cuda Toolkit: 10.1
- **GPU model and memory**: Tesla P100

I have written a custom CNN for a semantic segmentation problem. For the loss function, I have used 
tf.nn.softmax_cross_entropy_with_logits(labels=tf.cast(tf.squeeze(one_hot_labels),dtype=tf.float32),logits =tf.squeeze(pred))
Here is the interesting part, when some of my components within the image are labelled with values such as 100+(such as 105) the network fails to converge during training. If the same components are labelled with less than 100 such as (97) the loss converges. What could be the root cause of this issue here?

**Comment by gadagashwini** [comment]
@Raj-08 In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
